libs/wingsdk/src/shared-tf/tokens.ts

```diff
@@ -33,11 +38,16 @@ export class CdkTfTokens extends Tokens {
       throw new Error(`Tokens can only be bound by a Function for now`);
     }
 
-    const envName = this.envName(JSON.stringify(value));
+    const envName = this.envNameForToken(value);
     const envValue = Fn.jsonencode(value);
-    // the same token might be bound multiple times by different variables/inflight contexts
     if (host.env[envName] === undefined) {
       host.addEnvironment(envName, envValue);
     }
   }
+
+  private envNameForToken(value: any) {
+    const resolved = Tokenization.resolve(value, { scope: this.stack, resolver: new DefaultTokenResolver(new StringConcat()) });
+    return this.envName(resolved);
+  }
```

Comment:
Do you think there's a way to unit test this?

Reply:
yes, i'll look into this.